File: test-autoparams-parameters/src/test/java/test/autoparams/Product.java
package test.autoparams;

import java.math.BigDecimal;
import java.util.UUID;

import lombok.AllArgsConstructor;
import lombok.Getter;

@AllArgsConstructor
@Getter
public class Product {

    private final UUID id;
    private final String name;
    private final String imageUri;
    private final String description;
    private final BigDecimal priceAmount;
    private final int stockQuantity;
    private final boolean displayed;
}

File: test-autoparams-kotlin/src/test/kotlin/test/autoparams/Product.kt
package test.autoparams

import java.math.BigDecimal
import java.util.UUID

data class Product(
    val id: UUID,
    val name: String,
    val imageUri: String,
    val description: String,
    val priceAmount: BigDecimal,
    val stockQuantity: Int,
    val displayed: Boolean
)
